s3cortxutils/s3confstore/s3confstore/cortx_s3_confstore.py

```diff
@@ -26,138 +29,166 @@
 from cortx.utils.kv_store import kv_store_collection
 
 class S3CortxConfStore:
-  default_index = "default_cortx_s3_confstore_index"
 
-  def __init__(self):
-    """Instantiate constore"""
+  def __init__(self, config: str = None, index: str = "default_s3confstore_index"):
+    """Instantiate confstore."""
+    self.config_file = config
+    self.default_index = index
 
-  def load_config(self, index: str, backend_url: str):
-    """Load Config into constore"""
-    if not backend_url.strip():
-      print("Cannot load py-utils:confstore as backend_url[{}] is empty.".format(backend_url))
-      exit(-1)
-    else:
-      Conf.load(index, backend_url)
+    if config is not None:
+      self.validate_configfile(config)
+      self.load_config(self.default_index, self.config_file)
+
+  @staticmethod
+  def load_config(index: str = "default_s3confstore_index", config: str = ""):
+    """Load Config into confstore."""
+    Conf.load(index, config)
 
-  def get_config(self, index: str, key: str):
-    """Get the key's config from constore"""
-    return Conf.get(index, key)
+  def get_config(self, key: str):
+    """Get the key's config from confstore."""
+    return Conf.get(self.default_index, key)
 
-  def set_config(self, index: str, key: str, setval: str, save: bool = False):
-    """Set the key's value in constore"""
-    Conf.set(index, key, setval)
+  def set_config(self, key: str, value: str, save: bool = False):
+    """Set the key's value in confstore."""
+    Conf.set(self.default_index, key, value)
     if save == True:
-      """Update the index backend"""
-      Conf.save(index)
+      # Update the index backend.
+      Conf.save(self.default_index)
+
+  @staticmethod
+  def validate_configfile(configfile: str):
+    """Validate the 'configfile' url, if its a valid file and of supported format."""
+    if os.path.isfile(urlparse(configfile).path) != True:
+      print("config file: {} does not exist".format(configfile))
+      sys.exit(1)
+    else:
+      store_type = urlparse(configfile).scheme
+      is_valid_type = False
+      valid_types = ''
+      storage = inspect.getmembers(kv_store_collection, inspect.isclass)
+      for name, cls in storage:
+        if hasattr(cls, 'name') and name != "KvStore":
```

Comment:
see if we need to do case-sensitive comparison.

Reply:
@pratyush-seagate  - can you respond here ?

Reply:
This code checks the parsing capabilities provided by lower layers. KvStore needs to be avoided in this check. I don't think case-based checks are needed unless that class name changes.
Also this code is existing and works as expected.